homeassistant/components/generic_thermostat/climate.py

```diff
@@ -177,11 +205,25 @@ def __init__(
         self._max_temp = max_temp
         self._target_temp = target_temp
         self._unit = unit
-        self._support_flags = SUPPORT_FLAGS
-        if away_temp:
+
+        if presets and len(presets) > 0:
+            self._presets = presets
+        else:
+            self._presets = collections.OrderedDict()
+            if away_temp:
+                self._presets[PRESET_AWAY] = away_temp
+
+        self._presets[PRESET_NONE] = self._target_temp
+        if len(self._presets) > 1:
```

Comment:
This is a BC break aswell as you are only adding preset mode support when presets are defined.
Thats wrong as it already does support presets.

Reply:
`self._presets` are used to store `target_temp` as `PRESET_NONE`, new `presets` and legacy `away_temp`
If you don't use `away_temp` neither `presets` prests array contains only `PRESET_NONE`.
To have preset mode available you need at least two item in the array (`PRESET_NONE` and something else PRESET_*).
I can add some comment in the code for readability.